Validate Ansible 2.8 for release-3.11

mtnbikenc opened the pull request:
* Clean up bare variables
* openshift_facts.py - Clean up imports
* openshift_ca - Fix lib_utils filters not present on delegated tasks
* Health Checker - Stop using Ansible docker internals, execute docker command
* Fix master cert vars
* Allow and test with Ansible 2.8.0
* Fix library exception handler
* Remove unused group creation
* Refactor custom modules to not return `results`, which is a reserved var in Ansible 2.8

vrutkovs commented:
/test e2e-atomic
/test e2e-gcp-ha
/test e2e-gcp-logging
/test e2e-gluster

vrutkovs commented:
/test e2e-gluster

vrutkovs commented:
/test e2e-atomic

vrutkovs commented:
/test e2e-gcp-ha

vrutkovs commented:
/test e2e-gcp-logging

DanyC97 commented on roles/openshift_health_checker/library/docker_info.py:
```diff
@@ -1,24 +0,0 @@
-
-
-if __name__ == '__main__':
-    main()
```
YESSSS , thanks for cleaning up this file @mtnbikenc 

✔️

DanyC97 commented on roles/openshift_health_checker/openshift_checks/docker_storage.py:
```diff
@@ -42,22 +41,15 @@ class DockerStorage(DockerHostMixin, OpenShiftCheck):
-                    "msg": "Docker API query missing info:\n{}".format(json.dumps(docker_info))}
-        docker_info = docker_info["info"]
+        # attempt to get the docker info hash from the docker cli
+        command = ' '.join(['docker', 'info', '--format', '"{{json .}}"'])
```
wouldn't this fail when cri-o/ podman is installed ? or we saying that even with cri-o as container runtime, docker will be a must ?

mtnbikenc replied:
The `docker_storage` check is not run when `openshift_use_crio_only` is `True`.
https://github.com/openshift/openshift-ansible/blob/release-3.11/roles/openshift_health_checker/action_plugins/openshift_health_check.py#L73-L75

The original `docker_info` module was just getting docker info through Ansible internals.  I'm slowing removing as much reliance on Ansible internals as possible because it is a continual problem as Ansible development continues.

DanyC97 replied:
appreciate taking the time to respond @mtnbikenc ! keep up the excellent work 👍

vrutkovs commented:
/retest

vrutkovs commented:
```
[DEPRECATION WARNING]: The TRANSFORM_INVALID_GROUP_CHARS settings is set to 
allow bad characters in group names by default, this will change, but still be 
user configurable on deprecation. This feature will be removed in version 2.10.
 Deprecation warnings can be disabled by setting deprecation_warnings=False in 
ansible.cfg.
 [WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details

mtnbikenc commented:
@vrutkovs 
> ```
> [DEPRECATION WARNING]: The TRANSFORM_INVALID_GROUP_CHARS settings is set to 
> allow bad characters in group names by default, this will change, but still be 

I removed this group creation because it was no longer being used.
`oo_masters_deployment_type_{{ openshift_deployment_type }}`

Additional bare variables were cleaned up.  Rebased and squashed.

mtnbikenc commented:
Fixed logging vars and some additional `module_results` items.  Rebased and squashed.

jcpowermac commented on roles/lib_openshift/src/test/generate-and-run-tests.sh:
```diff
@@ -1,23 +1,22 @@
+#echo
+#echo "Running lib_utils generate-and-run-tests.sh"
+#echo "-------------------------------------------"
+#../../../lib_utils/src/test/generate-and-run-tests.sh
```
Is this supposed to be commented out?

mtnbikenc replied:
Thanks for the catch.  As I was running these tests, which are not part of our regular CI, I found that the lib_utils tests (for repoquery.py in particular) was broken on my Fedora system.  repoquery has different options between a yum based system and a dnf based system.  I'll see if I can come back to this and fix some more things.

mtnbikenc replied:
Tests and scripts are updated.

mtnbikenc commented:
More fixes.
/hold

mtnbikenc commented:
All unit and integration tests are passing.  Tested full install with all components manually with Ansible 2.8.0 and 2.7.5.  Grepped logs for deprecation warnings.
/hold cancel

```
INSTALLER STATUS ***********************************************************
Initialization                 : Complete (0:00:18)
Health Check                   : Complete (0:00:13)
Node Bootstrap Preparation     : Complete (0:03:54)
etcd Install                   : Complete (0:00:42)
Load Balancer Install          : Complete (0:00:14)
Master Install                 : Complete (0:05:34)
Master Additional Install      : Complete (0:01:14)
Node Join                      : Complete (0:00:33)
Hosted Install                 : Complete (0:00:38)
Cluster Monitoring Operator    : Complete (0:00:42)
Openshift Metering             : Complete (0:00:21)
Web Console Install            : Complete (0:00:18)
Console Install                : Complete (0:00:21)
metrics-server Install         : Complete (0:00:00)
Logging Install                : Complete (0:12:20)
Monitor Availability           : Complete (0:00:07)
Service Catalog Install        : Complete (0:03:18)
OLM Install                    : Complete (0:00:31)
Management Install             : Complete (0:11:02)
Descheduler Install            : Complete (0:00:08)
Node Problem Detector Install  : Complete (0:00:07)
Auto-heal Install              : Complete (0:00:13)
Cluster Auto Scaler Install    : Complete (0:00:13)
````

mtnbikenc commented:
/test logging

mtnbikenc commented:
/test logging

mtnbikenc commented:
/test logging

mtnbikenc commented:
Ansible 2.8.1 fixes an issue (in 2.8.0) when using `listen` on an `include_tasks` in a `handler`.  This affected the openshift_logging_elasticsearch handlers.

vrutkovs commented on roles/lib_openshift/src/test/integration/oc_process.yml:
```diff
@@ -66,13 +64,14 @@
-      - procout.results.results['items'][1]['metadata']['name'] == 'testdb'
-      - procout.results.results['items'][1]['kind'] == 'DeploymentConfig'
+      - procout.module_results.results['items'][0]['metadata']['name'] == 'testdb'
+      - procout.module_results.results['items'][0]['kind'] == 'Secret'
```
Its not clear why kind: Secret is being used here

mtnbikenc replied:
These integration tests had not been run in a long time.  The order of the results being returned changed from having 'Service' first to having 'Secret' returned.  This is only changing so items are in the order they are returned from the current version of the product, although it is not a 'failure' if they are in a different order.  This is why I added the comment on line 67.

vrutkovs commented on roles/lib_openshift/src/test/integration/oc_process.yml:
```diff
@@ -66,13 +64,14 @@
-      - procout.results.results['items'][1]['kind'] == 'DeploymentConfig'
+      - procout.module_results.results['items'][0]['metadata']['name'] == 'testdb'
+      - procout.module_results.results['items'][0]['kind'] == 'Secret'
+      - procout.module_results.results['items'][2]['metadata']['name'] == 'testdb'
```
Item 1 is being skipped?

mtnbikenc replied:
Same as above, but DeploymentConfig is now the third item.  More can be done to make these integration tests work better, I was just making sure that the data returned was correct and what would verify the module worked.

vrutkovs commented on roles/lib_utils/callback_plugins/aa_version_requirement.py:
```diff
@@ -22,7 +22,7 @@ def display(*args, **kwargs):
 # Set to minimum required Ansible version
 REQUIRED_VERSION = '2.5.7'
-DESCRIPTION = "Supported versions: %s or newer but less than 2.8.0" % REQUIRED_VERSION
+DESCRIPTION = "Supported versions: %s or newer" % REQUIRED_VERSION
```
Maybe its worth restricting 2.9.0 already, as it may have breaking changes?

mtnbikenc replied:
I'll add it if testing 2.9 RCs shows major problems.  I hope to be able to have anything fixed before 2.9 shipped.

vrutkovs commented on roles/lib_utils/library/delegated_serial_command.py:
```diff
@@ -192,9 +192,9 @@ def main():
-        except IOError as e:
-            if e.errno != errno.EAGAIN:
-                module.fail_json(msg="I/O Error {0}: {1}".format(e.errno, e.strerror))
+        except IOError as ioerror:
```
Is it just a cosmetic change or some bugfix?

mtnbikenc replied:
Yes, this was reporting a flake8 or pylint issue.

vrutkovs commented on roles/openshift_project_request_template/tasks/main.yml:
```diff
@@ -2,7 +2,7 @@
   command: mktemp /tmp/openshift-ansible-XXXXXX.yaml
   register: mktemp
-  changed_when: False
+  changed_when: False | bool
```
This change is probably not needed

mtnbikenc replied:
If there are not any other changes necessary to get this merged, I'd like to address this in a follow-up PR.

vrutkovs replied:
Sure

jcpowermac commented:
Completed install and upgrade using Ansible v2.6.17

mtnbikenc commented on playbooks/openshift-master/private/config.yml:
```diff
@@ -99,9 +99,6 @@
     import_tasks: tasks/enable_bootstrap.yml
 
   post_tasks:
```
The group created by `oo_masters_deployment_type_{{ openshift_deployment_type }}` is not in use in the code.  See https://github.com/openshift/openshift-ansible/pull/11325#issuecomment-498389719

jcpowermac commented:
/lgtm
